### PR TITLE
A tiny js world

### PR DESCRIPTION
**Pull request by yaripey**
### A tiny JS world

[Demo](https://yaripey.github.io/a-tiny-JS-world/)
[Code base](https://github.com/yaripey/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/yaripey/a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+inhabitants.forEach(elem => print(`${elem.species}; ${elem.name}; ${elem.gender}; ${elem.legs}; ${elem.hands}; ${elem.message}`))
```
Comment (OleksiyRudenko) [inline submissions/yaripey/a-tiny-js-world/index.js:65]: There si a pattern in the template string. How about using the power of `Array` methods like `map` and `join`?
Imagine you have to print dozens of props. Hardly you can avoid listing their names, but the rest would be less error prone and also you'd avoid a lengthy template string.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@yaripey good job!
Let's improve code readability, scalability and maintainability by employing `Array` methods.

File: submissions/yaripey/a-tiny-js-world/index.js (hunk 1 above)
Comment (yaripey) [reply submissions/yaripey/a-tiny-js-world/index.js:65]: Got it, will remake it, thanks.

File: submissions/yaripey/a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+inhabitants.forEach(elem => {
+  print(Object.keys(elem).map(property => elem[property]).join('; '))
```
Comment (OleksiyRudenko) [inline submissions/yaripey/a-tiny-js-world/index.js:65]: Please cross-check your solution implementations against task's requirements at all times

image

These exist for a reason.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@yaripey nice try.
However, the way you handled the pattern you had  is far away from what would be appropriate.
Consider that fragment agnostic of the task in general, focus on rersolving the issue with the pattern.

File: submissions/yaripey/a-tiny-js-world/index.js (hunk 2 above)
Comment (yaripey) [reply submissions/yaripey/a-tiny-js-world/index.js:65]: Omg, I completely missed that part of the task, thank you, will redo.

**Review by OleksiyRudenko — APPROVED** [review]
@yaripey great job!

**Pull request merged**